source/server/listener_impl.cc

```diff
@@ -467,6 +467,18 @@ void ListenerImpl::buildAccessLog() {
 void ListenerImpl::buildInternalListener() {
   if (config_.address().has_envoy_internal_address()) {
     internal_listener_config_ = std::make_unique<Network::InternalListenerConfig>();
+    if (config_.has_connection_balance_config() || config_.enable_mptcp() ||
```

Comment:
Does anything go wrong if this is thrown after `buildFilterChains` has already run? I can't remember if that has any interaction with the previous config on the listener. If there's any risk, you could move this into validateConfig() to do it earlier.

Reply:
As long as the exception was thrown prior to ListenerImpl::initialize(), there is no risk to wrongly deem the exception as a signal of "listener is wamed up".

We don't cross the line here

Reply:
api listener is handled early: Api listener config doesn't instantiate ListenerImpl but HttpApiListener
I add a test in listenerimpl pretending that some how a listenerImpl is constructed with Apilistener. Though in production it will never hit :)

Reply:
A separate question to @junr03: should we reject api listener with internal address? 
 https://github.com/envoyproxy/envoy/blob/main/source/server/api_listener_impl.h#L39